### ISSUE TITLE
no output when checking status of stratum 0 server

### ISSUE THREAD

**Issue by joeryan** [cvmfs-contrib/cvmfs-servermon#39]
I've configured a separate system to monitor a stratum 0 and multiple stratum 1 servers, however I get no output when checking the stratum 0 server.  stratum 1 servers return information normally.  I have tried installing the cvmfs-servermon package on the stratum 0 server directly, but it also outputs an empty json with format=details for any montests.   I would expect the check and whitelist to succeed.

**Comment by DrDaveD** [comment]
The check only shows output when there is a check status listed, because someone ran `cvmfs_server check -a`.  The whitelist should be there though.

**Comment by DrDaveD** [comment]
Does your `http://localhost/cvmfs/info/v1/repositories.json` list the stratum 0 repositories?  That file is automatically created by cvmfs_server so it should be there.

**Comment by joeryan** [comment]
Yes, the stratum 0 repositories are listed in the received repositories.json file.  However querying the servermon endpoint returns an empty json when run with montests=all or whitelist.

`[cvmfs-strat0 ~]$ curl localhost//cvmfs/info/v1/repositories.json                                                                      
{
  "schema" : 1,
  "repositories" : [
    {
      "name"  : "testing.icer.msu.edu",
      "url"   : "/cvmfs/testing.icer.msu.edu"
    }
  ],
  "replicas" : [
  ]
}
[cvmfs-strat0 ~]$ curl 'localhost/cvmfsmon/api/v1.0/all&format=details'                                                                                                   
{}
[cvmfs-strat0 ~]$ curl 'localhost/cvmfsmon/api/v1.0/whitelist&format=details'
{}
`

**Comment by DrDaveD** [comment]
I tried reproducing with a similar repositories.json and I got the expected result:
```
{"CRITICAL": {"whitelist": [{"msg": "whitelist is expired",
                             "repo": "test.gateway"}]}}
```

Are you using the latest cvmfs-servermon version, 1.22?

**Comment by joeryan** [comment]
Yes, it looks like I have that version installed from the cvmfs yum repo.
Version     : 1.22
Release     : 1.23.obs.el7

**Comment by DrDaveD** [comment]
I don't understand what's happening.  Is your stratum 0 port 80 open to the internet, or can you make it open to read requests from 131.225.0.0/16?  If so, tell me the DNS name of it.

What is the host operating system version?

**Comment by joeryan** [comment]
The host OS is CentOS 7.9, but unfortunately the server is not accessible from the internet.  It is strictly an internal resource and I don't have the ability to easily make any firewall changes or route to it from the internet.  I'll try to dig through and see what I can find about why I see the empty response.

**Comment by DrDaveD** [comment]
A nice thing about python is that you can insert debugging print statements even in binary distributions.  You should be able to hack the code in `/usr/share/cvmfs-servermon/webapi/*.py`.  Output goes to /var/log/httpd/error_log.  Reload httpd after modifying code to make it take effect.

**Comment by joeryan** [comment]
Thanks!  That's what I'll be doing.  I'll let you know of anything that I find.

**Comment by joeryan** [comment]
I realize that my version (2.9.4) is not the latest cvmfs-server package (2.10.0), so I wonder if that could be the issue.  My server does not return .cvmfs_status.json (not found) and that causes the whitelist to ignore them.  Any idea if the cvmfs_status.json was introduced after 2.9.4?  I don't see it in the 2.10 release notes, so I'm not sure.  I'll have to upgrade, but that might take me a little while.

**Comment by DrDaveD** [comment]
I wouldn't think that there would be that much of a difference from a slightly older cvmfs-server version like that.

**Comment by joeryan** [comment]
I wouldn't have expected that either, but tested it on an updated version anyway.  Even after creating a new test repository on a test server with the latest version, I still get a 404 error querying .cvmfs_status.json directly on the Stratum 0 server and an empty result when querying the cvmfsmon API endpoint.  I'm going to enable external access to my test stratum 0 server that shows this same behavior and maybe you can see something I'm missing.

**Comment by DrDaveD** [comment]
Ok let me know the URL when you are ready.

**Comment by joeryan** [comment]
A test  stratum 0 can now be reached at http://cvmfs.clanryan.us.

**Comment by DrDaveD** [comment]
Thanks for that, it enabled me to find and fix the bug.  Please update to cvmfs-servermon-1.23.
